freud/environment.pyx

```diff
@@ -681,8 +736,15 @@ cdef class EnvironmentMotifMatch(_MatchEnv):
 
     def compute(self, system, motif, threshold, neighbors=None,
                 registration=False):
-        r"""Determine clusters of particles that match the motif provided by
-        motif.
+        r"""Determine which particles have local environments
+            matching the given environment motif.
+
+        .. warning::
+
+            Comparisons between two environments are only made when both
+            environments contain the same number of particles.
+            However, this is not currently enforced, so no warning will be given
```

Comment:
Now I think it is confusing to say "However, this is not currently enforced" here. That makes it sound like it's not currently enforced that comparisons between two environments are only made when both environments contain the same number of particles. But that is enforced. I would just delete that part of the sentence and only say:

"No warning will be given at runtime if mismatched environments are provided for comparison to the motif."